### PR TITLE
Test for 2D zero padding.

### LINKED ISSUE #40724: Invalid test case from test_zero_padding_2d in convolutional_test.py
**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04

**Describe the current behavior**
https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/layers/convolutional_test.py#L654 
They try to test on different data_format configurations, ie. `channels_first`, `channels_last`. But for the code in the abovementioned link, it's meaningless, it always takes the last inputs. We should construct the inputs separately under different conditions. Even though it didn't cause any error but it should be fixed.

### PR DESCRIPTION
**Pull request by jpodivin**
I rewrote the unit test for 2D zero padding to support both channels first and channels last convention.

I used the `test_upsampling_2d(self)` as guideline to ensure consistency with other tests. 

Fixes #40724

### CODE REVIEW

**Comment by gbaned** [comment]
@jpodivin Can you please resolve conflicts? Thanks!

**Comment by jpodivin** [comment]
@gbaned 
The #41253 seems to have already fixed the issue, although in a slightly different fashion. Therefore I'm closing the PR.

**Comment by gbaned** [comment]
@jpodivin Thanks for confirmation, closing the PR.
